src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,257 @@
+    *         best divide the input volume. The second Seq contains the joints that remain from the input
+    *         joint set.
+    */
+  @tailrec
+  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block,
+                      numProcessors: Int, remainingJoints: Seq[Seq[Joint]] = Seq.empty[Seq[Joint]]):
+  (Seq[Joint], Seq[Joint]) = {
+    val allJoints = if (remainingJoints.isEmpty) {
+      jointSets
+    } else {
+      remainingJoints
+    }
+
+    if (jointSets.isEmpty) {
+      throw new IllegalStateException("ERROR: Unable to find satisfactory seed joints")
+    }
+    if (jointSets.head(0).shape.isEmpty) {
+      // Check that joint set is persistent
+      val seedJointOption = findSeedJoints(jointSets.head, inputVolume, numProcessors, inputVolume.volume)
+
+      // Check whether joint set yields satisfactory seed joints. If it does not, check next joint set.
+      if (seedJointOption.isEmpty) {
+          searchJointSets(jointSets.tail, inputVolume, numProcessors, allJoints)
+      } else {
+        val seedJoints = seedJointOption.get
+        val remainingFromJointSet = jointSets.head.diff(seedJoints)
+        val allOtherJoints = remainingJoints.take(remainingJoints.length - jointSets.length).flatten ++
+          remainingFromJointSet ++ jointSets.tail.flatten
+        (seedJoints, allOtherJoints)
+      }
+    } else {
+      searchJointSets(jointSets.tail, inputVolume, numProcessors, allJoints)
+    }
+  }
+
+  /**
+    * Cycles through persistent joint set to find joints that approximately balances load across processes
+    *
+    * @param jointSet Seq of Joints representing a persistent joint set
+    * @param rockVolume Block representing input rock volume that should be divided into approximately equal pieces
+    *                   for each process to operate on
+    * @param numProcessors Number of processes that will be used in analysis
+    * @param tolerance Tolerance in deviation from ideal volume per processor. Default initial value is 1%, though it
+    *                  will be updated as necessary in recursive calls.
+    * @param stepSize Step size that tolerance will be incremented by if unable to find enough seed joints. Default
+    *                 initial value is 1%, though it will be updated as necessary in recursive calls.
+    * @return If able to find the required number of seed joints, this function will return a Seq of Joints. If unable
+    *         to find sufficient seed joints, it will return None.
+    */
+  @tailrec
+  private def findSeedJoints(jointSet: Seq[Joint], rockVolume: Block, numProcessors: Integer, tolerance: Double = 0.01,
+                             stepSize: Double = 0.01): Option[Seq[Joint]] = {
+    if (jointSet.length < numProcessors - 1) {
+      throw new IllegalArgumentException(s"Error: Not enough joints to generate required number of seed joints. ${jointSet.length} joints but"+
```

Comment:
Do you think it's appropriate to throw an exception rather than return `None` in this case? Couldn't this function just be called again with a different joint set?